### PR TITLE
Added un-merged improvements from #9528

### PR DESCRIPTION
**Pull request by inoas**
I do still think that adding as inline comments the RFC and Section reference is highly helpful and also will make clear that we support the NON-STANDARD codes 499 and 599.

There are more than 16 RFCs ( https://github.com/cakephp/cakephp/pull/9528/files#diff-a57bc7407c64664db661cb46c465f363R36 ) and those won't change and its easy to discover what these then mean _by spec_. https://github.com/cakephp/cakephp/pull/9528/files#diff-a57bc7407c64664db661cb46c465f363R59

### CODE REVIEW

**Comment by inoas** [comment]
Ref: https://github.com/cakephp/cakephp/pull/9523

**Comment by jippi** [comment]
👎  to adding RFCs 
`meh` to changing the header texts - will probably make some apps sad, if they test for headers

**Comment by inoas** [comment]
* The texts now align with the official RFC texts.
* The status codes should be checked against, not the error texts.

**Comment by ADmad** [comment]
Tests failing

**Comment by inoas** [comment]
Yes the test assumes there are 64 status codes, but there are 65 now.

**Comment by ADmad** [comment]
> Yes the test assumes there are 64 status codes, but there are 65 now.

Thank you for the explanation, I couldn't have figured it out myself.

**Comment by inoas** [comment]
After closing my other PR, would you be so kind and fix that single test post-merge?

File: src/Network/Response.php [hunk 1]
```diff
@@ -56,7 +56,7 @@ class Response
         303 => 'See Other',
         304 => 'Not Modified',
         305 => 'Use Proxy',
-        306 => 'Switch Proxy',
+        306 => '(Unused)',
```
Comment (markstory) [inline src/Network/Response.php]: why is this here if its unused?

Comment (inoas) [inline src/Network/Response.php]: It has - as far as I can understand - been deprecated by the IETF - see my original https://github.com/cakephp/cakephp/pull/9528/files#diff-a57bc7407c64664db661cb46c465f363R85 especially https://tools.ietf.org/html/rfc2616#section-10.3.7

Comment (inoas) [inline src/Network/Response.php]: Also see https://tools.ietf.org/html/draft-cohen-http-305-306-responses-00

Comment (ADmad) [inline src/Network/Response.php]: So the question remains why add it to the list when this code is not supposed to be used?

Comment (inoas) [inline src/Network/Response.php]: I can remove it. However I can't see the full picture if that's a good thing to do but rather keep it as it is. It is often not omitted but simply set to 'Unused', sometimes even the old message is there and its not clear that it should not be used.

It doesn't - however - exist on https://httpstatuses.com.
If you agree (just thumb up) I can remove 306.

Comment (dereuromark) [inline src/Network/Response.php]: I would keep it as it was for 3.x - we can re-discuss this for 4.x.

Comment (inoas) [inline src/Network/Response.php]: I hope this is in a mergable state now.

**Comment by dereuromark** [comment]
Maybe you could squash history (single commit).

**Comment by markstory** [comment]
[Tests are failing](https://travis-ci.org/cakephp/cakephp/jobs/163389549#L746)

**Comment by inoas** [comment]
Added test. Squashed commits.

**Comment by dereuromark** [comment]
Let's ship it then.
